src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -4040,26 +4259,321 @@ ColocationGroupCreateCommandList(void)
+		Datum forceDelegationDatum =
+			heap_getattr(nextTuple,
+						 Anum_pg_dist_object_force_delegation,
+						 pgDistObjectDesc,
+						 &forceDelegationIsNull);
+		bool forceDelegation = DatumGetBool(forceDelegationDatum);
+
+		if (distributionArgumentIndexIsNull)
+		{
+			distributionArgumentIndex = INVALID_DISTRIBUTION_ARGUMENT_INDEX;
+		}
+
+		if (colocationIdIsNull)
+		{
+			colocationId = INVALID_COLOCATION_ID;
+		}
+
+		if (forceDelegationIsNull)
+		{
+			forceDelegation = NO_FORCE_PUSHDOWN;
+		}
+
+		char *workerMetadataUpdateCommand =
+			MarkObjectsDistributedCreateCommand(list_make1(address),
+												list_make1_int(distributionArgumentIndex),
+												list_make1_int(colocationId),
+												list_make1_int(forceDelegation));
+		SendCommandListToActivatedNodes(context, list_make1(workerMetadataUpdateCommand));
 	}
+	MemoryContextSwitchTo(oldContext);
 
-	appendStringInfo(colocationGroupCreateCommand,
-					 ") SELECT pg_catalog.citus_internal_add_colocation_metadata("
-					 "colocationid, shardcount, replicationfactor, "
-					 "distributioncolumntype, coalesce(c.oid, 0)) "
-					 "FROM colocation_group_data d LEFT JOIN pg_collation c "
-					 "ON (d.distributioncolumncollationname = c.collname "
-					 "AND d.distributioncolumncollationschema::regnamespace"
-					 " = c.collnamespace)");
+	systable_endscan_ordered(pgDistObjectScan);
+	index_close(pgDistObjectIndexRel, AccessShareLock);
+	relation_close(pgDistObjectRel, NoLock);
+}
+
+
+/*
+ * SendInterTableRelationshipCommands sends inter-table relationship commands
+ * (e.g. constraints, attach partitions) to workers with transactional or
+ * nontransactional mode per inter table relationship according to transactionMode
+ * inside metadataSyncContext.
+ */
+void
+SendInterTableRelationshipCommands(MetadataSyncContext *context)
+{
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 0;
+
+	Relation relation = table_open(DistPartitionRelationId(), AccessShareLock);
+
+	TupleDesc tupleDesc = RelationGetDescr(relation);
+	SysScanDesc scanDesc = systable_beginscan(relation,
+											  InvalidOid, false,
+											  NULL, scanKeyCount, scanKey);
+
+	MemoryContext oldContext = MemoryContextSwitchTo(context->context);
+	HeapTuple nextTuple = NULL;
+	while (true)
+	{
+		if (!InMetadataSyncTestMode(context))
+		{
+			MemoryContextReset(context->context);
+		}
+
+		nextTuple = systable_getnext(scanDesc);
+
+		if (!HeapTupleIsValid(nextTuple))
+		{
+			break;
+		}
+
+		Oid relationId = FetchRelationIdFromPgPartitionHeapTuple(nextTuple, tupleDesc);
+
+		if (!ShouldSyncTableMetadata(relationId))
+		{
+			continue;
+		}
+
+		/*
+		 * skip foreign key and partition creation when the Citus table is
+		 * owned by an extension
+		 */
+		if (IsTableOwnedByExtension(relationId))
+		{
+			continue;
+		}
+
+		List *commandList = InterTableRelationshipOfRelationCommandList(relationId);
+
+		if (commandList == NIL)
+		{
+			continue;
+		}
+
+		commandList = lcons(DISABLE_DDL_PROPAGATION,
```

Comment:
no need new line

(I'll not write these comments, can you please go over the patch for indentation once more)